### PR TITLE
add surface hint for recon/reference/raw

### PR DESCRIPTION
**Pull request by XinfengZhang**
Signed-off-by: Carl Zhang <carl.zhang@intel.com>

### CODE REVIEW

**Comment by XinfengZhang** [comment]
@fulinjie @dvrogozh please help to review

File: va/va.h [hunk 1]
```diff
@@ -876,6 +880,20 @@ typedef struct _VAConfigAttrib {
 /** \brief Driver supports base mode for slice decoding */
 #define VA_DEC_SLICE_MODE_BASE         0x00000002
 
+/** @name Attribute values for VAConcigAttribValSurfaceRes */
+/**@{*/
+typedef union _VAConcigAttribValSurfaceReq{
```
Comment (wangyan-intel) [inline va/va.h]: Config?

File: va/va.h [hunk 2]
```diff
@@ -876,6 +880,20 @@ typedef struct _VAConfigAttrib {
...
+    struct {
+    /** \brief encode raw input surface and recon surface is with different layout inside backend driver
+     *  default value is zero, when it is 1, application need to set surface usage hint
+     *  /c VA_SURFACE_ATTRIB_USAGE_HINT_RECON & VA_SURFACE_ATTRIB_USAGE_HINT_RAW when call vaCreateSurfaces to create surfaces */
+    uint32_t input_recon_diff :1;
+    /** \brief Reserved for future use. must be zero */
+    uint32_t reserved:31;
+    }bits;
+    uint32_t value;
+}VAConcigAttribValSurfaceReq;
```
Comment (wangyan-intel) [inline va/va.h]: Same.

File: va/va.h [hunk 3]
```diff
@@ -777,6 +777,10 @@ typedef enum
      * implementation, multiple frames encode/decode can improve HW concurrency
      */
     VAConfigAttribMultipleFrame         = 40,
+    /** \brief this attribute is used to check addtional surface requirement
+     * the default value is 0, other value need to checkk /c VAConcigAttribValSurfaceReq
```
Comment (fulinjie) [inline va/va.h]: Also, s/VAConcig/VAConfig

File: va/va.h [hunk 4]
```diff
@@ -876,6 +880,20 @@ typedef struct _VAConfigAttrib {
 /** \brief Driver supports base mode for slice decoding */
 #define VA_DEC_SLICE_MODE_BASE         0x00000002
 
+/** @name Attribute values for VAConfigAttribValSurfaceRes */
```
Comment (fulinjie) [inline va/va.h]: NIT: 
s/SurfaceRes/SurfaceReq

**Comment by HeJunyan** [comment]
Information is enough.
I want to have a try before it merge, it that ready in iHD driver now?

**Comment by fulinjie** [comment]
Here is some comments from the community:
http://ffmpeg.org/pipermail/ffmpeg-devel/2020-March/258119.html

The basic question is why  not handling it transparently (and not to introduce extra attribute).
Hi @XinfengZhang, would you please help to elaborate whether its' possible?

**Comment by XinfengZhang** [comment]
to make it transparent, it will bring a lot of effort, 1. need to record the real format information inside driver. 2. do not create real surface until it is was used, it is difficult to implement it, we we call vaCreateSurfaces, we even don't know the context, config...

**Comment by XinfengZhang** [comment]
close it for https://github.com/intel/media-driver/pull/915

**Pull request closed without merging**